File: eternal_return/src/main/java/org/EternalReturn/ERCharacter/Character/yuki/Character_Yuki.java
package org.EternalReturn.ERCharacter.Character.yuki;


import org.EternalReturn.ERCharacter.ERCharacter;
import org.EternalReturn.ERPlayer.ERPlayer;
import org.bukkit.Location;
import org.bukkit.Particle;
import org.bukkit.Sound;
import org.bukkit.entity.Entity;
import org.bukkit.entity.LivingEntity;
import org.bukkit.entity.Player;
import org.bukkit.util.Vector;

import java.util.*;

public class Character_Yuki extends ERCharacter {

    public int buttonCount = 4;
    public boolean isReloading = false;
    public boolean isActiveSkill = false;
    public Map<LivingEntity, Location> stunnedVictims = new HashMap<>();

    public Character_Yuki(ERPlayer player) {
        super(player);
        this.ActiveCooldownSeconds = 3;
        this.PassiveCooldownSeconds = 8;
        registerMonobehaviour(new Active());
        registerMonobehaviour(new Passive());
    }


    @Override
    public String getName() { return "yuki"; }

}

File: eternal_return/src/main/kotlin/org/eternalreturn/ercharacter/character/yuki/Character_Yuki.kt
package org.eternalreturn.ercharacter.character.yuki


import org.bukkit.entity.Player
import org.eternalreturn.ercharacter.CooldownContext
import org.eternalreturn.ercharacter.datastructure.CoolTableSeconds
import org.eternalreturn.ercharacter.datastructure.DamageTable
import org.eternalreturn.erplayer.ERPlayer
import org.eternalreturn.system.EREngine

class Character_Yuki(erEngine : EREngine, player: Player) : ERPlayer(player, erEngine) {
    var buttonCount: Int = 4
    var isReloading: Boolean = false
    var isActiveSkill: Boolean = false


    override val activeCoolForEachLevel  = CoolTableSeconds(this::activeLevel,  30, 27, 24, 21, 18, 0)
    override val passiveCoolForEachLevel = CoolTableSeconds(this::passiveLevel, 7, 6, 5, 4, 3, 0)

    val activeCooldownContext = CooldownContext(activeCoolForEachLevel);

    val passiveExtraDamageForEachLevel   = DamageTable(this::activeLevel, 1.0, 2.0 ,3.0, 4.0, 5.0, 100.0)
    val activeExtraDamageForEachLevel    = DamageTable(this::passiveLevel,3.0, 4.0, 5.0, 6.0, 8.0, 100.0)

    init {
        this.ActiveCooldownSeconds = 3
        this.PassiveCooldownSeconds = 8
        registerMonobehaviour(Active(activeCooldownContext));
        registerMonobehaviour(Passive())
    }


    override val name: String
        get() = "yuki"
}